packages/design-system-react-native/src/components/Button/variants/ButtonTertiary/ButtonTertiary.stories.tsx

```diff
@@ -0,0 +1,183 @@
+  title: 'Components/ButtonTertiary',
+  component: ButtonTertiary,
+  argTypes: {
+    children: {
+      control: 'text',
+    },
+    size: {
+      control: 'select',
+      options: ButtonBaseSize,
+    },
+    isLoading: {
+      control: 'boolean',
+    },
+    loadingText: {
+      control: 'text',
+    },
+    startIconName: {
+      control: 'select',
+      options: IconName,
+    },
+    endIconName: {
+      control: 'select',
+      options: IconName,
+    },
+    isDisabled: {
+      control: 'boolean',
+    },
+    isFullWidth: {
+      control: 'boolean',
+    },
+    isDanger: {
+      control: 'boolean',
+    },
+    isInverse: {
+      control: 'boolean',
+    },
+    twClassName: {
+      control: 'text',
+    },
+  },
+};
+
+export default meta;
+
+type Story = StoryObj<ButtonTertiaryProps>;
+const ButtonTertiaryStory: React.FC<ButtonTertiaryProps> = ({
+  isInverse,
+  ...props
+}) => {
+  const tw = useTailwind();
+  return (
+    <View
+      style={[
+        tw`${isInverse ? `bg-primary-default` : 'bg-background-default'}`,
+      ]}
+    >
+      <ButtonTertiary isInverse={isInverse} {...props} />
+    </View>
+  );
+};
+
+export const Default: Story = {
+  args: {
+    children: 'Sample ButtonTertiary Text',
+    size: DEFAULT_BUTTONTERTIARY_PROPS.size,
+    isLoading: DEFAULT_BUTTONTERTIARY_PROPS.isLoading,
+    loadingText: 'Loading',
+    startIconName: IconName.Add,
+    endIconName: IconName.AddSquare,
+    isDisabled: DEFAULT_BUTTONTERTIARY_PROPS.isDisabled,
+    isFullWidth: DEFAULT_BUTTONTERTIARY_PROPS.isFullWidth,
+  },
+  render: (args) => <ButtonTertiaryStory {...args} />,
+};
+
+export const Sizes: Story = {
+  render: () => (
+    <View style={{ gap: 16 }}>
+      <ButtonTertiary size={ButtonBaseSize.Sm}>ButtonSize Sm</ButtonTertiary>
+      <ButtonTertiary size={ButtonBaseSize.Md}>ButtonSize Md</ButtonTertiary>
+      <ButtonTertiary size={ButtonBaseSize.Lg}>
+        ButtonSize Lg (Default)
+      </ButtonTertiary>
+    </View>
+  ),
+};
+
+export const IsLoading: Story = {
+  render: () => (
+    <View style={{ gap: 16 }}>
+      <ButtonTertiary isLoading>ButtonTertiary</ButtonTertiary>
+      <ButtonTertiary isLoading loadingText="With Loading Text">
+        ButtonTertiary
+      </ButtonTertiary>
+    </View>
+  ),
+};
+
+export const WithStartAccessory: Story = {
+  render: () => (
+    <ButtonTertiary startIconName={IconName.Add}>ButtonTertiary</ButtonTertiary>
+  ),
+};
+
+export const WithEndAccessory: Story = {
+  render: () => (
+    <ButtonTertiary endIconName={IconName.Add}>ButtonTertiary</ButtonTertiary>
+  ),
+};
+
+export const WithStartAndEndAccessory: Story = {
+  render: () => (
+    <ButtonTertiary
+      startIconName={IconName.Add}
+      endIconName={IconName.AddSquare}
+    >
+      ButtonTertiary
+    </ButtonTertiary>
+  ),
+};
+
+export const isDisabled: Story = {
+  render: () => <ButtonTertiary isDisabled>ButtonTertiary</ButtonTertiary>,
+};
```

Comment:
non-blocking suggestion: consider using `args` instead of `render` for better Storybook performance. `args` enables component caching, efficient prop updates, and maintains control functionality.